### DISCUSSION TITLE
Using [asar-plus] can still extract app.asar

### DISCUSSION THREAD

**Discussion by getliuyue** [sleeyax/asarmor#75] [General]
I have updated your asarmor@3.0.0 Test using commands: asar-plus e -iff -iu app.asar ./app/
You can still extract the app.asar to ./app, is there any way to prevent asar-plus

https://github.com/evlon/asar-plus

**Comment by sleeyax** [comment]
Have you enabled encryption? Starting from v3 we will focus more on that, other protection methods are no longer as effective as they used to be.

**Comment by getliuyue** [comment]
Yes, i use the following code
const asarmor = require('asarmor');
const { join } = require("path");

exports.default = async ({ appOutDir, packager }) => {
  try {
    const asarPath = join(packager.getResourcesDir(appOutDir), 'app.asar');
    console.log(`asarmor applying patches to ${asarPath}`);
    const archive = await asarmor.open(asarPath);
    archive.patch(); // apply default patches
    await archive.write(asarPath);
  } catch (err) {
    console.error(err);
  }
};

And I have enabled encryption, using normal unpacking methods will result in multiple 1GB sized files, but using asar-plus will directly result in normal files

**Comment by sleeyax** [comment]
You're still missing something in your code. Please see https://github.com/sleeyax/asarmor/blob/f9a2c148312df481de47ef0a6b62e527aa1ddd8e/example/node/build.js#L11 and https://github.com/sleeyax/asarmor/blob/f9a2c148312df481de47ef0a6b62e527aa1ddd8e/example/electron/afterPack.js#L13 as examples.

It seems like `asar-plus` was specifically designed to combat asarmor's base protections so encryption is the only thing that can protect your asar file contents from it.

**Comment by getliuyue** [comment]
Sorry, I didn't read the document carefully, but I am using `vue-cli-plugin-electron-builder`, which is different from the structure in the document. How should I configure `beforePack.js` and where should I copy main.node to?

**Comment by getliuyue** [comment]
I have enabled encryption now, and all the JavaScript files packaged into the `app.asar` have been encrypted by AES, but I don't know how to automatically decrypt them at runtime

**Comment by sleeyax** [comment]
> Sorry, I didn't read the document carefully, but I am using `vue-cli-plugin-electron-builder`, which is different from the structure in the document. How should I configure `beforePack.js` and where should I copy main.node to?

I can only provide support for electron builder for now, it's impossible to support all plugins / build tools. If this plugin is based on electron builder, you should be able to follow the guide along in a similar manner while also checking the documentation.

**Comment by sleeyax** [comment]
> I have enabled encryption now, and all the JavaScript files packaged into the `app.asar` have been encrypted by AES, but I don't know how to automatically decrypt them at runtime

For this you need to copy the *.node files and change your entry point in package.json, as shown in the readme and examples.
